lib/Onyx.js

```diff
@@ -918,26 +933,30 @@ function hasPendingMergeForKey(key) {
  * @returns {Promise}
  */
 function set(key, value) {
-    if (_.isNull(value)) {
+    let valueCopy = value;
```

Comment:
This variable name is misleading because it's not a copy yet, but still a reference to the same value. It doesn't become an actual copy until `cleanObject(valueCopy)` is ran. My suggestion is to remove this line, change `if (_.isNull(value)) {` and then `const valueCopy = cleanObject(value);`

Reply:
+1 

With a NAB, I still don't love `valueCopy`. it's not really a copy at all since we modify the object and remove the null properties. Maybe `valueWithNullRemoved`. Sorry if that's too Clean Code™️ I swear I am not a Java developer 😄